### PR TITLE
ci: Add ZE build steps to Github Action

### PR DESCRIPTION
**Pull request by aingerson**
Signed-off-by: aingerson <alexia.ingerson@intel.com>

### CODE REVIEW

**Comment by aingerson** [comment]
@rajachan

File: .github/workflows/pr-ci.yml [hunk 1]
```diff
@@ -81,7 +81,11 @@ jobs:
       - name: Install Xe
```
Comment (acgoldma) [inline .github/workflows/pr-ci.yml:81]: I would change this to "Ze (oneAPI Level Zero).

Comment (aingerson) [reply .github/workflows/pr-ci.yml]: Updated

**Comment by aingerson** [comment]
@acgoldma @rajachan Could you help me understand the failure? Seems consistent.
Also, I didn't realize the provider enable configure options used their arguments:
```
  OFI_PROVIDER_FLAGS: >-
    --enable-efa=rdma-core/build
    --enable-mrail
    --enable-psm3=rdma-core/build
    --enable-rxd
    --enable-rxm
    --enable-shm
    --enable-tcp
    --enable-udp
    --enable-usnic
    --enable-verbs=rdma-core/build
```
Is that proper usage of the enable arguments? I don't see verbs, efa, or psm3 using it as far as I can tell. Is there a reason they are being set here?

**Comment by acgoldma** [comment]
Issue seems to be with libibverbs:
```
checking if libibverbs is linkable by libtool... no
```
It used to be "yes"

**Comment by rajachan** [comment]
> Is that proper usage of the enable arguments? I don't see verbs, efa, or psm3 using it as far as I can tell. Is there a reason they are being set here?

The config opts looks right to me. That's how the providers using rdma-core take a custom path today. Without it, it will try to find rdma-core in system paths (and fail if not installed, given the explicit --enable-provider flag).

That said, taking a path with --enable args instead of --with args is a bit icky, but that something we'd have to change uniformly across our configury.

**Comment by aingerson** [comment]
Yeah I'm just trying to understand what could be causing it to not be able to find verbs, whether it has something to do with the the arguments or the stuff I added. Any ideas?

**Comment by acgoldma** [comment]
> Yeah I'm just trying to understand what could be causing it to not be able to find verbs, whether it has something to do with the the arguments or the stuff I added. Any ideas?

I think we need to get the config.log out from the machine.
Maybe add a step to cat that on failure?
I am not sure if there is a way to archive/save files on failure?

**Comment by rajachan** [comment]
Does a manual build with the same flags work? `config.log` would have more meaningful information. Is the verbs driver Ze-aware? If so, this could actually be exposing a bug in that configure. Btw, your fork of the repo would have these actions running when you push to a new branch as well, so you can debug with a test branch pushed to your fork with any debug code without affecting this PR.

**Comment by acgoldma** [comment]
https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts

looks like there is an action: `actions/upload-artifact@v2`

Also Stack Overflow :)
https://stackoverflow.com/questions/64037059/upload-artifact-in-github-action

**Comment by aingerson** [comment]
> Btw, your fork of the repo would have these actions running when you push to a new branch as well, so you can debug with a test branch pushed to your fork with any debug code without affecting this PR.

@rajachan Awesome. I'll experiment there. How do I view these tests/results?

**Comment by rajachan** [comment]
Should be under https://github.com/aingerson/libfabric/actions/.

**Comment by aingerson** [comment]
That gives me a 404. Do I need to do something to set it up?

**Comment by acgoldma** [comment]
> That gives me a 404. Do I need to do something to set it up?

https://github.com/aingerson/libfabric/settings/actions
Make sure Actions are enabled

**Comment by rajachan** [comment]
Here's the failure:

```
configure:23961: result: yes
configure:24407: checking if libibverbs is linkable by libtool
configure:24418: ./libtool --mode=link --tag=CC gcc   -Iyes/include  -I/usr/local/cuda/include   -Irdma-core/build/include -O2 -DNDEBUG -fvisibility=hidden -Wall -Wundef -Wpointer-arith conftemp.15237.c -o conftemp   -Lyes/lib64  -L/usr/local/cuda/lib64   -Lrdma-core/build/lib -libverbs
./libtool: line 7563: cd: yes/lib64: No such file or directory
libtool:   error: cannot determine absolute directory name of 'yes/lib64'
configure:24426: result: no
configure: failed program was
char ibv_open_device ();
int main ()
{ return ibv_open_device (); }
```

Looks like --with-ze configure logic is using "yes" as the path when an explicit path is not provided.

I took the liberty of adding a commit on top of yours to generate config.log artifacts for failed builds.

**Comment by aingerson** [comment]
@rajachan Thanks! That commit is super helpful. I'll leave that in to merge for future debugging!
Yes, the issue is definitely the flags not getting set correctly with the "yes" case and I've been working a fix for it but can't figure out what the flags are getting set to so I pulled in your commit and it's not creating the artifact. I can see it fine on the PR build but not the one on my branch. Do you know if I have to set something to enable artifacts? Or does it live in a different location?

**Comment by rajachan** [comment]
I see it in your fork too: https://github.com/aingerson/libfabric/actions/runs/1035494682

**Comment by aingerson** [comment]
Ahh it's in a different location. I didn't scroll all the way down. Thanks!

**Comment by aingerson** [comment]
Updated with configure.ac fix for --with-ze argument. I'm sure there is a better way to do this - I don't deal a lot with autotools... but this works. I tried putting it inside the FI_CHECK_PACKAGE do on success arg but for some reason it didn't work and so this is my workaround for that. Open to feedback though.
@rajachan I think CUDA might need to be updated as well? I think the same thing would happen if the application sets flags themselves and then passes in --with-cuda without a path

**Comment by rajachan** [comment]
I thought I fixed this for CUDA already with https://github.com/ofiwg/libfabric/pull/6294, but let me know if you are still seeing issues.

**Pull request merged**